src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -541,10 +575,41 @@ public void startJenkins() throws Throwable {
         // TODO options to set Winstone options, etc.
         // TODO pluggable launcher interface to support a Dockerized Jenkins JVM
         // TODO if test JVM is running in a debugger, start Jenkins JVM in a debugger also
+        pb.redirectErrorStream(true);
         proc = pb.start();
-        // TODO prefix streams with per-test timestamps & port
-        new StreamCopyThread(description.toString(), proc.getInputStream(), System.out).start();
-        new StreamCopyThread(description.toString(), proc.getErrorStream(), System.err).start();
+        OutputStream out = System.out;
+        if (name != null) {
+            out = new LineTransformationOutputStream.Delegating(out) {
+                @Override protected void eol(byte[] b, int len) throws IOException {
+                    out.write('[');
+                    out.write(name.getBytes(StandardCharsets.US_ASCII));
+                    out.write(']');
+                    out.write(' ');
+                    if (color != null) {
+                        out.write(27); // ESC
+                        out.write('[');
+                        out.write(color.code.getBytes(StandardCharsets.US_ASCII));
+                        out.write('m');
+                        int preNlLen = len;
+                        while (preNlLen > 0 && (b[preNlLen - 1] == '\n' || b[preNlLen - 1] == '\r')) {
+                            preNlLen--;
+                        }
+                        assert 0 <= preNlLen;
+                        assert preNlLen <= len;
+                        assert len <= b.length;
+                        out.write(b, 0, preNlLen);
+                        out.write(27);
+                        out.write('[');
+                        out.write('0');
+                        out.write('m');
+                        out.write(b, preNlLen, len - preNlLen);
+                    } else {
+                        out.write(b, 0, len);
+                    }
+                }
+            };
```

Comment:
I don't suppose it would be worth extracting this into a static class and writing a unit test for some common cases (e.g., regular line, empty line, Windows line break, etc.).

Reply:
> extracting this into a static class

Or just a `static` method. Yes, sounds prudent.